### PR TITLE
adding the created_at check to reduce the loading time

### PR DESCRIPTION
**Pull request by 9to1url**

### CODE REVIEW

**Comment by 9to1url** [comment]
I refactored to fit the PR. please review and accept.

File: lib/blogit/models/post/meta.ex [hunk 1]
```diff
@@ -120,12 +121,27 @@ defmodule Blogit.Models.Post.Meta do
...
-    author = data["author"] || file_info[:author]
-    author = if author == "", do: "Anonymous", else: author
+    
+
+    created_at = data["created_at"]
+    updated_at = data["updated_at"]
+    author = data["author"]
+    {created_at2, updated_at2, author2} =
+      if created_at == nil do
+        IO.inspect "get from REPO using git log:" <> file_path
```
Comment (meddle0x53) [inline lib/blogit/models/post/meta.ex]: Use the `Logger` and not `IO` for logging.

File: lib/blogit/models/post/meta.ex [hunk 2]
```diff
@@ -120,12 +121,27 @@ defmodule Blogit.Models.Post.Meta do
...
-    updated_at = data["updated_at"] || file_info[:updated_at]
-    author = data["author"] || file_info[:author]
-    author = if author == "", do: "Anonymous", else: author
+    
+
+    created_at = data["created_at"]
+    updated_at = data["updated_at"]
+    author = data["author"]
+    {created_at2, updated_at2, author2} =
+      if created_at == nil do
```
Comment (meddle0x53) [inline lib/blogit/models/post/meta.ex]: Only when the `created_at` is `nil`?
Shouldn't it be `if is_nil(created_at) || is_nil(updated_at) || is_nil(author)  do`?
Basically you have to set all three of them to not ask the provider.

Also use `is_nil` to check for `nil`s.

File: lib/blogit/models/post/meta.ex [hunk 3]
```diff
@@ -120,12 +121,27 @@ defmodule Blogit.Models.Post.Meta do
...
-    created_at = data["created_at"] || file_info[:created_at]
-    updated_at = data["updated_at"] || file_info[:updated_at]
-    author = data["author"] || file_info[:author]
-    author = if author == "", do: "Anonymous", else: author
+    
+
+    created_at = data["created_at"]
+    updated_at = data["updated_at"]
+    author = data["author"]
+    {created_at2, updated_at2, author2} =
```
Comment (meddle0x53) [inline lib/blogit/models/post/meta.ex]: I prefer naming them in a way we know where they came from:
`created_at_from_provider` for example and similar naming for the others.

File: lib/blogit/models/post/meta.ex [hunk 4]
```diff
@@ -120,12 +121,27 @@ defmodule Blogit.Models.Post.Meta do
...
+      if created_at == nil do
+        IO.inspect "get from REPO using git log:" <> file_path
+        file_info = repository.provider.file_info(repository.repo, path)
+        created_at3 = file_info[:created_at] || "2020-01-26T01:01:01"
```
Comment (meddle0x53) [inline lib/blogit/models/post/meta.ex]: Use `DateTime.utc_now() |> DateTime.to_iso8601()` for the default. Don't hardcode dates.
Also don't name them here, just return:
```
{
  file_info[:created_at] || DateTime.utc_now() |> DateTime.to_iso8601(),
  updated_at || file_info[:updated_at] || DateTime.utc_now() |> DateTime.to_iso8601(),
  author || file_info[:author] || "Anonymous"
}
```

File: lib/blogit/models/post/meta.ex [hunk 5]
```diff
@@ -120,12 +121,27 @@ defmodule Blogit.Models.Post.Meta do
...
+        updated_at3 = updated_at || file_info[:updated_at] || created_at3
+        author3 = author || file_info[:author] || "Anonymous"
+        {created_at3, updated_at3, author3}
+      else
+        {nil,nil,nil}
```
Comment (meddle0x53) [inline lib/blogit/models/post/meta.ex]: Add spaces between `nil,` and the next `nil`.
We should introduce the formatter to this project.

**Review by meddle0x53 — CHANGES_REQUESTED** [review]
Your idea to not call the provider if we already have the data is right.
I have a few requests, mainly code-style related.

Also you forgot to bump the `mix.exs` version (at least I don't see the bump in the PR)

**Comment by 9to1url** [comment]
fixed all the issues you mentioned in code review. thanks.

**Review by meddle0x53 — APPROVED** [review]
Very cool!
Thanks a lot!

**Pull request merged**